### PR TITLE
Annotate for nullable reference types

### PR DESCRIPTION
**Pull request by NextTurn**
Left about 130 warnings for you to consider, where null-forgiving is involved. :)

Feel free to push my branch for any additional changes.

### CODE REVIEW

**Comment by leculver** [comment]
Awesome work.  I'll work through these remaining warnings today.

File: src/Microsoft.Diagnostics.Runtime/src/Builders/RuntimeBuilder.cs [hunk 1]
```diff
@@ -345,7 +349,7 @@ IReadOnlyList<ClrAppDomain> IRuntimeHelpers.GetAppDomains(ClrRuntime runtime, ou
             int i = 0;
             foreach (ulong domain in domainList)
             {
-                ClrAppDomain? ad = GetOrCreateAppDomain(builder, domain);
+                ClrAppDomain ad = GetOrCreateAppDomain(builder, domain);
                 if (ad != null)
```
Comment (NextTurn) [inline src/Microsoft.Diagnostics.Runtime/src/Builders/RuntimeBuilder.cs]: No need to check for a non-nullable value.

File: src/Microsoft.Diagnostics.Runtime/src/Utilities/Symbols/DefaultSymbolLocator.Async.cs [hunk 2]
```diff
@@ -125,18 +125,17 @@ private async Task<string> SearchSymbolServerForFile(string fileSimpleName, stri
...
             }
 
-            string result = await GetFirstNonNullResult(tasks).ConfigureAwait(false);
+            string? result = await GetFirstNonNullResult(tasks).ConfigureAwait(false);
             return result;
         }
 
-        private static async Task<T> GetFirstNonNullResult<T>(List<Task<T>> tasks)
-            where T : class
+        private static async Task<string> GetFirstNonNullResult(List<Task<string?>> tasks)
```
Comment (NextTurn) [inline src/Microsoft.Diagnostics.Runtime/src/Utilities/Symbols/DefaultSymbolLocator.Async.cs]:
```suggestion
        private static async Task<string?> GetFirstNonNullResult(List<Task<string?>> tasks)
```

There are codes checking against null for the call chain:

```
FindBinaryAsync
DownloadFileWorker
SearchSymbolServerForFile
GetFirstNonNullResult
```

https://github.com/microsoft/clrmd/blob/19725d98eae81948c7c9dbbfc85cdd65217cc636/src/Microsoft.Diagnostics.Runtime/src/Utilities/Symbols/DefaultSymbolLocator.Async.cs#L54

Comment (leculver) [reply src/Microsoft.Diagnostics.Runtime/src/Utilities/Symbols/DefaultSymbolLocator.Async.cs]: Thanks!

I am going to leave this for later.  Going through the symbol locator code is on my todo list so I'm hesitant to do more cleanups there until I know what the final result will look like.

**Pull request merged**